src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/WebhookConfiguration.java

```diff
@@ -75,54 +85,109 @@ public WebhookConfiguration(@CheckForNull final String committersToIgnore) {
+            }
+
+            // Handle null case
+            String thisCommittersToIgnore = committersToIgnore;
+            if (thisCommittersToIgnore == null) {
+                thisCommittersToIgnore = "";
+            }
+
+            if (!hookCommittersToIgnore.trim().equals(thisCommittersToIgnore.trim())) {
+                serverHook.setCommittersToIgnore(committersToIgnore);
+                return true;
+            }
+
+            return false;
+        }
+
+        if (hook instanceof NativeBitbucketServerWebhook) {
+            boolean updated = false;
+
+            NativeBitbucketServerWebhook serverHook = (NativeBitbucketServerWebhook) hook;
+            String url = getNativeServerWebhookUrl(owner.getServerUrl());
+
+            if (!url.equals(serverHook.getUrl())) {
+                serverHook.setUrl(url);
+                updated = true;
+            }
+
+            List<String> events = serverHook.getEvents();
+            if (events == null) {
+                serverHook.setEvents(NATIVE_SERVER_EVENTS);
+                updated = true;
+            } else if (!events.containsAll(NATIVE_SERVER_EVENTS)) {
+                Set<String> newEvents = new TreeSet<>(events);
+                newEvents.addAll(NATIVE_SERVER_EVENTS);
+                serverHook.setEvents(new ArrayList<>(newEvents));
+                updated = true;
+            }
+
+            return updated;
         }
-        return hook;
+
+        return false;
     }
-    
-    public BitbucketWebHook getHook(BitbucketSCMSource owner) {
-        if (BitbucketCloudEndpoint.SERVER_URL.equals(owner.getServerUrl())) {
+
+    BitbucketWebHook getHook(BitbucketSCMSource owner) {
+        final String serverUrl = owner.getServerUrl();
+        if (BitbucketCloudEndpoint.SERVER_URL.equals(serverUrl)) {
             BitbucketRepositoryHook hook = new BitbucketRepositoryHook();
             hook.setEvents(CLOUD_EVENTS);
             hook.setActive(true);
             hook.setDescription(description);
             hook.setUrl(Jenkins.getActiveInstance().getRootUrl() + BitbucketSCMSourcePushHookReceiver.FULL_PATH);
             return hook;
-        } else {
-            BitbucketServerWebhook hook = new BitbucketServerWebhook();
-            hook.setActive(true);
-            hook.setDescription(description);
-            hook.setUrl(Jenkins.getActiveInstance().getRootUrl() + BitbucketSCMSourcePushHookReceiver.FULL_PATH);
-            hook.setCommittersToIgnore(committersToIgnore);
-            return hook;
         }
+
+        switch (BitbucketServerEndpoint.findWebhookImplementation(serverUrl)) {
+            case NATIVE: {
+                NativeBitbucketServerWebhook hook = new NativeBitbucketServerWebhook();
+                hook.setActive(true);
+                hook.setEvents(NATIVE_SERVER_EVENTS);
+                hook.setDescription(description);
+                hook.setUrl(getNativeServerWebhookUrl(serverUrl));
+                return hook;
```

Comment:
so odd not to read break 🤔

Reply:
Just personal coding preference. I can change those to use breaks if you prefer.